wp-sw-cache/lib/service-worker.js

```diff
@@ -1,103 +1,127 @@
+      return Promise.all(Object.keys(this.urls).map(function(url) {
+        var hash = wpSwCache.urls[url];
 
-    event.respondWith(
-      caches.match(lookupRequest)
-        .then(function(response) {
-          if (response) {
-            if (debug) {
-              console.log('[fetch] Returning from ServiceWorker cache: ', event.request.url);
-            }
-            return response;
+        // ... get its hash from storage ...
+        return wpSwCache.storage.getItem(url).then(function(value) {
+          // ... and if nothing has changed, just move on to the next URL
+          if(value === hash) {
+            log('[install] Hash unchanged, doing nothing: ', url);
+            return Promise.resolve();
           }
-          if (debug) {
-            console.error('[fetch] Cache miss! This should not happen. It implies problems caching.');
+          else {
+            log('[install] Hash changed or new URL, adding to cache "' + storageKey +'" : ' , url);
+            // ... Add the new/updated URL and its response to cache ...
+            return self.caches.open(storageKey).then(function(cache) {
+              return cache.add(url).then(function() {
+                // ... and once it's successful add its hash to storage
+                return wpSwCache.storage.setItem(url, hash);
+              });
+            });
           }
-          return fetch(event.request);
-        }
-      )
-    );
-  });
-
-  function normalizeAndAnonymize(request) {
-    var url = new URL(request.url);
-    if (url.origin !== location.origin) {
-      return request;
-    }
-
-    url.search = '';
-    url.fragment = '';
-    return new Request(url, {
-      method: request.method,
-      headers: request.headers,
-      mode: 'no-cors',
-      credentials: request.credentials,
-      cache: request.cache,
-      redirect: request.redirect,
-      referrer: request.referrer,
-      integrity: request.integrity
-    });
-  }
-
-  function shouldBeHandled(request) {
-    return request.method === 'GET' && CACHE_FILES.indexOf(request.url) !== -1;
-  }
-
-  self.addEventListener('activate', function(event) {
-    if (debug) {
-      console.log('[activate] Activating ServiceWorker!');
-    }
+        });
+      }));
+    },
+    // Check each URL in cache and delete anything that shouldn't be there anymore
+    // i.e. the user unchecked a file's box in admin
+    removeOldUrls: function() {
+      return caches.open(storageKey).then(function(cache) {
+        return cache.keys().then(function(keys) {
+          return Promise.all(keys.map(function(key) {
+            if(!(key.url in wpSwCache.urls)) {
+              log('[removeOldUrls] Removing URL no longer desired: ', key.url);
+              return cache.delete(key);
+            }
+            return Promise.resolve();
+          }));
+        })
+      });
+    },
+    // Install step that kicks off adding/updating URLs in cache and storage
```

Comment:
For the sake of clarity, you could move these two method to the top of the object.